bullet_train-super_load_and_authorize_resource/app/controllers/concerns/bullet_train/loads_and_authorizes_resource.rb

```diff
@@ -202,22 +201,14 @@ def account_load_and_authorize_resource(model, options, old_options = {})
     end
   end
 
-  def regex_to_remove_controller_namespace
-    raise "This is a template method that needs to be implemented by controllers including LoadsAndAuthorizesResource."
-  end
-
   def load_team
-    # Not all objects that need to be authorized belong to a team,
-    # so we give @team a nil value if no association is found.
-    begin
-      # Sometimes `@team` has already been populated by earlier `before_action` steps.
-      @team ||= @child_object&.team || @parent_object&.team
-    rescue NoMethodError
-      @team = nil
-    end
+    @team ||= @child_object&.try(:team) || @parent_object&.try(:team)
 
-    # Update current attributes.
-    Current.team = @team
+    return unless @team
+
+    if defined?(Current) && Current.respond_to?(:team=)
+      Current.team = @team
+    end
 
     # If the currently loaded team is saved to the database, make that the user's new current team.
     if @team.try(:persisted?)
```

Comment:
Since we've inserted the `return unless @team` above, we can remove the `try` here:

```suggestion
    if @team.persisted?
```

Reply:
I'm not calling `try` for null safety. I'm calling it because there is nothing in this gem to guarantee that the value of `@team` responds to `persisted?`

Reply:
Seems like it'd be better to raise the `NoMethodError` exception with our expected interface to ease debugging, but this code path was already wrapped in `try` so we can keep it.